File: 2706.buy-two-chocolates.cpp
// @leet start
class Solution {
public:
  int buyChoco(vector<int> &prices, int money) {
    int choc1 = INT_MAX;
    int choc2 = choc1;
    for (int price : prices) {
      if (price < choc1) {
        choc2 = choc1;
        choc1 = price;
      } else if (price < choc2) {
        choc2 = price;
      }
    }
    int answer = money - (choc1 + choc2);

    return answer >= 0 ? answer : money;
  }
};
// @leet end

File: 2706.buy-two-chocolates.py
# @leet start
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        choc1 = inf
        choc2 = choc1
        for price in prices:
            if price < choc1:
                choc2 = choc1
                choc1 = price
            elif price < choc2:
                choc2 = price

        total = money - (choc1 + choc2)
        if total < 0:
            return money
        return total


# @leet end
